src/DIRAC/Interfaces/Utilities/DCommands.py

```diff
@@ -0,0 +1,658 @@
+
+        return S_OK()
+
+    def hasProfile(self, profile):
+        return self.config.has_section(profile)
+
+    def defaultProfile(self):
+        retVal = self.get("global", "default_profile")
+        if not retVal["OK"]:
+            return None
+        return retVal["Value"]
+
+    def sections(self):
+        return self.config.sections()
+
+    def items(self, section):
+        return self.config.items(section)
+
+    def existsOrCreate(self, section, option, value):
+        if self.config.has_section(section) and self.config.has_option(section, option):
+            return False
+        self.set(section, option, value)
+        return True
+
+    def fillMinimal(self):
+        defaultGroup = ""
+        # Try to find the default user group
+        # todo: dirac-proxy-info contains the user group, use this
+        resultInfo = ProxyInfo.getProxyInfo()
+        if resultInfo["OK"]:
+            userName = resultInfo["Value"].get("username")
+            if userName:
+                result = Registry.findDefaultGroupForUser(userName)
+                if result["OK"]:
+                    defaultGroup = result["Value"]
+            if not defaultGroup:
+                defaultGroup = resultInfo["Value"].get("group")
+        if not defaultGroup:
+            defaultGroup = "dirac_user"
+        modified = False
+        modified |= self.existsOrCreate("global", "default_profile", defaultGroup)
+        modified |= self.existsOrCreate(defaultGroup, "group_name", defaultGroup)
+        modified |= self.existsOrCreate(defaultGroup, "home_dir", "/")
+        modified |= self.existsOrCreate(defaultGroup, "default_se", "DIRAC-USER")
+
+        return modified
+
+
+def createMinimalConfig(configDir=os.path.expanduser("~/.dirac"), configFilename="dcommands.conf"):
+
+    dconfig = DConfig(configDir, configFilename)
+
+    modified = dconfig.fillMinimal()
+
+    if modified:
+        dconfig.write()
+
+
+# -------------------------
+# DCommands Session helpers
+# -------------------------
+
+
+class DSession(DConfig):
+    __ENV_SECTION = "session:environment"
+
+    @classmethod
+    def sessionFilePrefix(cls):
+        return f"dsession.{uuid.getnode():x}"
+
+    @classmethod
+    def sessionFilename(cls, pid):
+        return cls.sessionFilePrefix() + ".%d" % (pid,)
+
+    def __init__(self, profileName=None, config=None, sessionDir=None, pid=None):
+        self.origin = config or DConfig()
+        modified = self.origin.fillMinimal()
+        if modified:
+            self.origin.write()
+
+        self.pid = pid
+        if not self.pid:
+            if "DCOMMANDS_PPID" in os.environ:
+                self.pid = int(os.environ["DCOMMANDS_PPID"])
+            else:
+                self.pid = os.getppid()
+
+        if not sessionDir:
+            var = "DCOMMANDS_SESSION_DIR"
+            if var in os.environ:
+                sessionDir = os.environ[var]
+            else:
+                sessionDir = self.origin.configDir
```

Comment:
These env variables...? Why? And where are they defined?

Reply:
I think the idea was that COMDIRAC has it's own ecosystem and does not interfere with anything else. Changing this now would break it for existing users; this would also be something for a more long term development. @atsareg: Any comment on this ?